src/wasm-binary.h

```diff
@@ -473,6 +473,9 @@ class WasmBinaryWriter : public Visitor<WasmBinaryWriter, void> {
   WasmBinaryWriter(Module* input, BufferWithRandomAccess& o, bool debug) : wasm(input), o(o), debug(debug) {
     prepare();
   }
+  WasmBinaryWriter(Module& input, BufferWithRandomAccess& o, bool debug) : wasm(&input), o(o), debug(debug) {
```

Comment:
A constructor accepting a reference instead of a pointer is required for WebIDL use (in src/js/binaryen.idl)